compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -373,34 +388,106 @@ void RecordMinMax::profileData(const std::string &mode, const std::string &input
-        _interpreter->writeInputTensor(input_node, input_data.data(), input_data.size());
+        vector_input_data[i].emplace_back(std::move(input_data));
       }
-
-      _interpreter->interpret();
     }
 
-    std::cout << "Recording finished. Number of recorded data: " << num_records << std::endl;
+    return vector_input_data;
   }
   catch (const H5::Exception &e)
   {
     H5::Exception::printErrorStack();
     throw std::runtime_error("HDF5 error occurred.");
   }
+}
+
+void RecordMinMax::profileData(const std::string &mode, const std::string &input_data_path,
+                               float min_percentile, float max_percentile)
+{
+  const auto input_nodes = loco::input_nodes(_module->graph());
+  const auto num_inputs = input_nodes.size();
+  const auto vector_input_data = importH5Data(input_data_path);
+  const auto num_records = vector_input_data.size();
+
+  for (int32_t record_idx = 0; record_idx < num_records; record_idx++)
+  {
+    std::cout << "Recording " << record_idx << "'th data" << std::endl;
+
+    for (int32_t input_idx = 0; input_idx < num_inputs; input_idx++)
+    {
+      const auto *input_node = loco::must_cast<const luci::CircleInput *>(input_nodes[input_idx]);
+      const auto &cur_input_data = vector_input_data[record_idx][input_idx];
+
+      // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
+      //       We can redcue the copy by directly writing data from file to interpreter inputs
+      _interpreters[0]->writeInputTensor(input_node, cur_input_data.data(), cur_input_data.size());
+    }
+
+    _interpreters[0]->interpret();
+  }
+
+  std::cout << "Recording finished. Number of recorded data: " << num_records << std::endl;
+
+  update_quantparam(_observers[0].get(), mode, min_percentile, max_percentile);
+}
+
+void RecordMinMax::profileData_with_parallel_record(const std::string &mode,
+                                                    const std::string &input_data_path,
+                                                    float min_percentile, float max_percentile)
+{
+  assert(_interpreters.size() == _threads_size);
+  assert(_observers.size() == _threads_size);
+  const auto vector_input_data = importH5Data(input_data_path);
+  const auto num_records = vector_input_data.size();
+  const auto input_nodes = loco::input_nodes(_module->graph());
+  const auto num_inputs = input_nodes.size();
+
+#pragma omp parallel for
+  for (uint32_t record_idx = 0; record_idx < num_records; record_idx++)
+  {
+    const auto current_thread_num = omp_get_thread_num();
+    for (int32_t input_idx = 0; input_idx < num_inputs; input_idx++)
+    {
+      const auto *input_node = loco::must_cast<const luci::CircleInput *>(input_nodes[input_idx]);
+
+      // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
+      //       We can reduce the copy by directly writing data from file to interpreter inputs
```

Comment:
The comments can be removed.

```suggestion
      // TODO: Input data is copied twice (file -> buffer (input_data) -> interpreter inputs)
      //       We can reduce the copy by directly writing data from file to interpreter inputs
```